decoder/expr_function.go

```diff
@@ -55,6 +56,61 @@ func (fe functionExpr) CompletionAtPos(ctx context.Context, pos hcl.Pos) []lang.
 
 		prefix := rootName[0:prefixLen]
 		return fe.matchingFunctions(prefix, eType.Range())
+
+	case *hclsyntax.ExprSyntaxError:
+		// TODO: "provider:" does not jump here, seems to be some other expression?
```

Comment:
Is this too edge-casey? Do we need a follow-up issue for this?

Reply:
If you want, you could do a quick check to see if you can catch the `ScopeTraversalExpr` case, and if it gets too complicated, create a follow-on issue. But I'm fine with creating a follow-on issue right away.

Reply:
Created a follow-up issue: https://github.com/hashicorp/hcl-lang/issues/385